common/datavalues/src/data_value_ops.rs

```diff
@@ -29,29 +29,48 @@ impl DataValue {
     }
 
     pub fn try_from_literal(literal: &str) -> Result<DataValue> {
-        match literal.parse::<i64>() {
+        let mut s = literal;
+        let negative_flag = s.starts_with(char::from_u32(45).unwrap());
+        if negative_flag {
+            s = &s[1..];
+        }
+
+        match s.parse::<u64>() {
             Ok(n) => {
-                if n >= 0 {
-                    let n = literal.parse::<u64>()?;
-                    if n <= u8::MAX as u64 {
-                        return Ok(DataValue::UInt8(Some(n as u8)));
-                    } else if n <= u16::MAX as u64 {
-                        return Ok(DataValue::UInt16(Some(n as u16)));
-                    } else if n <= u32::MAX as u64 {
-                        return Ok(DataValue::UInt32(Some(n as u32)));
+                if negative_flag {
+                    // when the real number is i8::MIN or i16::MIN or i32::MIN or i64::MIN
```

Comment:
if it's zero, use u8;

if it's negative, use i64 and check the ranges.

if it's positive, use u64 and check the ranges.

refer: https://github.com/ClickHouse/ClickHouse/blob/f2374a6916295c236abe9029d0c5b8cbeb3c8d9b/src/DataTypes/FieldToDataType.cpp

Reply:
Better to simplify this.

Reply:
ok